census/vanilla-tf/trainer/task.py

```diff
@@ -0,0 +1,334 @@
+  (age, education_num, capital_gain, capital_loss, hours_per_week,
+   gender, race, education, marital_status, relationship, workclass,
+   occupation, native_country) = input_columns
+
+  # bucketize the age feature
+  # See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/ops/bucketization_op.py
+  age_bucket = bucketization_op.bucketize(age, [18, 25, 30, 35, 40, 45, 50, 55, 60, 65])
+
+  wide_columns = [
+      sparse_cross([education, occupation], 15 * 16, name='edu_occ'),
+      sparse_cross(
+        [age_bucket, race, occupation],
+        10 * 5 * 15,
+        name='age_race_occ'),
+      sparse_cross(
+        [native_country, occupation],
+        42 * 15,
+        name='native_country_occ'),
+      gender,
+      native_country,
+      education,
+      occupation,
+      workclass,
+      marital_status,
+      relationship,
+      age_bucket
+  ]
+
+  return wide_columns
+
+
+def concat_wide_columns(wide_columns):
+  """Concat the tensors from wide columns."""
+
+  (edu_occ, age_race_occ, native_country_occ, gender,
+   native_country, education, occupation, workclass,
+   marital_status, relationship, age_bucket) = wide_columns
+
+  dense_tensors = [
+      sparse_to_dense(edu_occ, 15 * 16),
+      sparse_to_dense(gender, 2),
+      sparse_to_dense(workclass, 9),
+      sparse_to_dense(native_country, 42),
+      sparse_to_dense(education, 16),
+      sparse_to_dense(occupation, 15),
+      sparse_to_dense(workclass, 9),
+      sparse_to_dense(marital_status, 7),
+      sparse_to_dense(relationship, 6)
+  ]
+
+  return tf.concat(dense_tensors, 1)
+
+
+def sparse_to_dense(sparse_tensor, vocab_size):
+  """Convert the sparse to dense tensor."""
+  dense_tensor = tf.sparse_to_indicator(sparse_tensor, vocab_size)
+  dense_tensor = tf.cast(dense_tensor, tf.int32)
+  return dense_tensor
+
+def read_input_tensor(input_file, skiprows=None):
+  """Concatenate the wide columns to produce a single tensor."""
+  inp, label = read_input_data(input_file, skiprows)
+  in_tensor, label_tensor = generate_input(inp, label)
+  return concat_wide_columns(generate_wide_columns(in_tensor)), label_tensor
+
+def loss(model, labels):
+  """Compute cross entropy loss function."""
+  cross_entropy = tf.reduce_mean(
+      tf.nn.softmax_cross_entropy_with_logits(logits=model, labels=labels))
+  return cross_entropy
+
+def optimizer(loss, global_step):
+  """Gradient descent optimizer with 0.5 learning rate."""
+  train_step = tf.train.GradientDescentOptimizer(0.5).minimize(
+      loss, global_step=global_step)
+  return train_step
+
+
+def training_steps(session, train_step, model, labels, global_step, max_steps,
+                   inp, label,
+                   eval_inp, eval_label):
+  """Run the training steps and calculate accuracy every 10 steps."""
+  step = tf.train.global_step(session, global_step)
+
+  while step < max_steps:
+    session.run(
```

Comment:
Where is your batching? Do you really want to do Full gradient descent here? Shouldn't we demonstrate batching?

Maybe add `tf.train.shuffle_batch` to your `read_input_tensor` method.